components/nimbus/src/stateless/cirrus_client.rs

```diff
@@ -67,21 +65,31 @@ impl Default for EnrollmentRequest {
     fn default() -> Self {
         Self {
             client_id: None,
-            app_context: Default::default(),
             request_context: Default::default(),
             is_user_participating: true,
-            next_experiments: Default::default(),
             prev_enrollments: Default::default(),
         }
     }
 }
 
 #[derive(Default)]
-pub struct CirrusClient {}
+pub struct InternalMutableState {
+    experiments: Vec<Experiment>,
+}
+
+#[derive(Default)]
+pub struct CirrusClient {
+    app_context: AppContext,
+    state: Mutex<InternalMutableState>,
+}
 
 impl CirrusClient {
-    pub fn new() -> Self {
-        Self {}
+    pub fn new(app_context: String) -> Self {
+        let app_context: AppContext = from_str(&app_context).unwrap();
```

Comment:
```suggestion
    pub fn new(app_context: String) -> Result<Self> {
        let app_context: AppContext = from_str(&app_context)?;
```

A panic should really only be used for unusual errors that are neither caller nor callees fault (e.g. out of memory, or malformed installation, corrupted memory), rather than "you used me wrong".